### DISCUSSION TITLE
Building a REST API around Whatsapp4j

### DISCUSSION THREAD

**Discussion by jain-jigar** [Auties00/WhatsappWeb4j#257] [Q&A] [answered]
Hello,

I am trying to create REST API around Whatsapp4j  using Spring boot (v 2.7.10). No matter what I try I am getting the following error
```
java.lang.UnsupportedClassVersionError: Preview features are not enabled for it/auties/whatsapp/api/Whatsapp (class file version 61.65535). Try running with '--enable-preview'
``` 

I am using VS Code as my editor and have set settings.json as
```
{
    "java.configuration.runtimes": [
        {
          "name": "JavaSE-17",
          "path": "C:\\Program Files\\Eclipse Adoptium\\jdk-17.0.6.10-hotspot",
          "default": true
        },
    ],
 
}
``` 

My POM.XML is
```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>2.7.10</version>
		<relativePath/> <!-- lookup parent from repository -->
	</parent>
	<groupId>com.example</groupId>
	<artifactId>demo</artifactId>
	<version>0.0.1-SNAPSHOT</version>
	<name>demo</name>
	<description>Demo project for Spring Boot</description>
	<properties>
		<java.version>17</java.version>
	</properties>
	<dependencies>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter</artifactId>
		</dependency>

		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-test</artifactId>
			<scope>test</scope>
		</dependency>

		<dependency>
    		<groupId>com.github.auties00</groupId>
    		<artifactId>whatsappweb4j</artifactId>
    		<version>3.0-RC28</version>
		</dependency>

		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-web</artifactId>
		</dependency>

	</dependencies>

	<build>
		<plugins>
			<plugin>
				<groupId>org.springframework.boot</groupId>
				<artifactId>spring-boot-maven-plugin</artifactId>
				<configuration>
                    <source>17</source>
                    <target>17</target>
                    <release>${java.version}</release>
                    <compilerArgument>--enable-preview</compilerArgument>
                    <compilerArgs>--enable-preview</compilerArgs>
                </configuration>
			</plugin>
		</plugins>
	</build>

</project>

```

maven version 3.8.4
Java version: 17.0.6
OS name: "windows 10"

Any help is appreciated.

**Comment by Auties00** [comment, marked as answer]
> Hello,
> 
> I am trying to create REST API around Whatsapp4j using Spring boot (v 2.7.10). No matter what I try I am getting the following error
> 
> ```
> java.lang.UnsupportedClassVersionError: Preview features are not enabled for it/auties/whatsapp/api/Whatsapp (class file version 61.65535). Try running with '--enable-preview'
> ```
> 
> I am using VS Code as my editor and have set settings.json as
> 
> ```
> {
>     "java.configuration.runtimes": [
>         {
>           "name": "JavaSE-17",
>           "path": "C:\\Program Files\\Eclipse Adoptium\\jdk-17.0.6.10-hotspot",
>           "default": true
>         },
>     ],
>  
> }
> ```
> 
> My POM.XML is
> 
> ```
> <?xml version="1.0" encoding="UTF-8"?>
> <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
> 	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
> 	<modelVersion>4.0.0</modelVersion>
> 	<parent>
> 		<groupId>org.springframework.boot</groupId>
> 		<artifactId>spring-boot-starter-parent</artifactId>
> 		<version>2.7.10</version>
> 		<relativePath/> 
> 	</parent>
> 	<groupId>com.example</groupId>
> 	<artifactId>demo</artifactId>
> 	<version>0.0.1-SNAPSHOT</version>
> 	<name>demo</name>
> 	<description>Demo project for Spring Boot</description>
> 	<properties>
> 		<java.version>17</java.version>
> 	</properties>
> 	<dependencies>
> 		<dependency>
> 			<groupId>org.springframework.boot</groupId>
> 			<artifactId>spring-boot-starter</artifactId>
> 		</dependency>
> 
> 		<dependency>
> 			<groupId>org.springframework.boot</groupId>
> 			<artifactId>spring-boot-starter-test</artifactId>
> 			<scope>test</scope>
> 		</dependency>
> 
> 		<dependency>
>     		<groupId>com.github.auties00</groupId>
>     		<artifactId>whatsappweb4j</artifactId>
>     		<version>3.0-RC28</version>
> 		</dependency>
> 
> 		<dependency>
> 			<groupId>org.springframework.boot</groupId>
> 			<artifactId>spring-boot-starter-web</artifactId>
> 		</dependency>
> 
> 	</dependencies>
> 
> 	<build>
> 		<plugins>
> 			<plugin>
> 				<groupId>org.springframework.boot</groupId>
> 				<artifactId>spring-boot-maven-plugin</artifactId>
> 				<configuration>
>                     <source>17</source>
>                     <target>17</target>
>                     <release>${java.version}</release>
>                     <compilerArgument>--enable-preview</compilerArgument>
>                     <compilerArgs>--enable-preview</compilerArgs>
>                 </configuration>
> 			</plugin>
> 		</plugins>
> 	</build>
> 
> </project>
> ```
> 
> maven version 3.8.4 Java version: 17.0.6 OS name: "windows 10"
> 
> Any help is appreciated.

You have to both compile and run with --enable-preview
Your build config looks right, the run one is probably wrong

**Comment by nexa-backend** [comment]
coul you give the example building rest api cobalt using java spring boot ?
